start.sh

```diff
@@ -66,43 +78,68 @@ fi
 #create start script for X11
 echo "#!/bin/bash" > /home/chromium/xstart.sh
 
-# Only for the Pi4 since setting `display_rotate=1` doesn't work for KMS
-# rotate screen if env variable is set [normal, inverted, left or right]
-if [[ ! -z "$ROTATE_DISPLAY" ]]; then
-  echo "(sleep 3 && xrandr -o $ROTATE_DISPLAY) &" >> /home/chromium/xstart.sh
-fi
-
 # if no window size has been specified, find the framebuffer size and use that
 if [[ -z ${WINDOW_SIZE+x} ]]
   then
     export WINDOW_SIZE=$( cat /sys/class/graphics/fb0/virtual_size )
     echo "Using fullscreen: $WINDOW_SIZE"
+  else
+    echo "Custom window size set to $WINDOW_SIZE"
+fi
+
+# if no window position has been specified, use the default
+if [[ -z ${WINDOW_POSITION+x} ]]
+  then
+    export WINDOW_POSITION="0,0"
+    echo "Using default window position"
+fi
+
+# rotate screen if env variable is set [normal, inverted, left or right]
+if [[ ! -z "$ROTATE_DISPLAY" ]]; then
+  echo "(sleep 3 && xrandr -o $ROTATE_DISPLAY) &" >> /home/chromium/xstart.sh
+
+  #If the display is rotated to left or right, we need to reverse the size and position coords
+  if [[ "$ROTATE_DISPLAY" == "left" ]] || [[ "$ROTATE_DISPLAY" == "right" ]]; then
+    
+    echo "Display rotated to portait. Reversing screen coordinates"
+    
+    #window size
+    REVERSED_SIZE="$(reverse_window_coordinates $WINDOW_SIZE)"
+    WINDOW_SIZE=$REVERSED_SIZE
+    echo "Reversed window size: $WINDOW_SIZE"
+    
+    #window position, if set
+    if [[ "$WINDOW_POSITION" -ne "0,0" ]]
+    then
+      REVERSED_POSITION="$(reverse_window_coordinates $WINDOW_POSITION)"
+      export WINDOW_POSITION=$REVERSED_POSITION
+      echo "Reversed window position: $WINDOW_POSITION"
+    fi
+  fi
 fi
 
 # Set whether to run Chromium in config mode or not
 # This sets the cursor to show by default
 if [ ! -z ${KIOSK+x} ] && [ "$KIOSK" -eq "1" ]
   then
-    export KIOSK='--kiosk --start-fullscreen'
     echo "Enabling kiosk mode"
-    export CHROME_LAUNCH_URL="--app=$LAUNCH_URL"
-
-    #Set whether to show a cursor or not
-    if [[ ! -z $SHOW_CURSOR ]] && [[ "$SHOW_CURSOR" -eq "1" ]]
-      then
-        export CURSOR=''
-        echo "Enabling cursor"
-      else
-        export CURSOR='-- -nocursor'
-        echo "Disabling cursor"
-    fi
+    export CHROME_LAUNCH_URL="--app=$LAUNCH_URL" 
   else
-    export KIOSK=''
     export CHROME_LAUNCH_URL="$LAUNCH_URL"
     export CURSOR=''
     echo "Enabling cursor"
 fi
```

Comment:
Maybe we shouldn't modify cursor if we are not in kiosk mode  since we now allow to modify the cursor visibility independently in this PR

considering a generic use case of browser block and given kiosk mode is not the default, we can show the cursor by default

Reply:
Yep - good catch, I missed that line. It might be quicker for you to edit it, since I'm tied up today.

Reply:
@rahul-thakoor ---^

Reply:
ok. I will do that